src/Disks/ObjectStorages/HDFS/registerDiskHDFS.cpp

```diff
@@ -50,6 +52,22 @@ void registerDiskHDFS(DiskFactory & factory)
             DiskType::HDFS,
             /* send_metadata = */ false,
             copy_thread_pool_size);
+
+#ifdef NDEBUG
+        bool use_cache = true;
+#else
+        /// Current S3 cache implementation lead to allocations in destructor of
+        /// read buffer.
+        bool use_cache = false;
```

Comment:
What bad can happen in the case of allocations?

Reply:
I don't remember... I suppose we have some additional checks somewhere in debug mode...